### DISCUSSION TITLE
Ignoring %ignore for a rule?

### DISCUSSION THREAD

**Discussion by jnwatson** [lark-parser/lark#934] [Q&A] [answered]
If I have 
```
grammar = r'''
SPACE: " "
%ignore SPACE
start:  foo bar
foo: "foo"
bar: "bar"
'''

LarkParser = lark.Lark(grammar, parser='lalr')
text = 'foo bar'
tree = LarkParser.parse(text)
print(tree)
```

This parses fine. What if in this case, I don't want to allow a space between the foo and bar rules? Is there a way to ignore the ignore directive for (just one) rule?

**Comment by MegaIng** [comment, marked as answer]
No. You can however define a Terminal like this:

```

start: FOO_BAR

FOO_BAR: FOO BAR
```

But then `FOO` and `BAR` have to be terminals. Inside a terminal `%ignore` is not applied.

**Reply by MegaIng** [reply]
There was however an issue some time ago proposing this: #629

**Reply by jnwatson** [reply]
Thank you. I definitely need something along the lines that #629 proposes.
